tests/ship_streamer.cpp

```diff
@@ -66,18 +67,35 @@ int main(int argc, char* argv[]) {
       {
          boost::beast::flat_buffer abi_buffer;
          stream.read(abi_buffer);
-         std::string abi_string((const char*)abi_buffer.data().data(), abi_buffer.data().size());
-         eosio::json_token_stream token_stream(abi_string.data());
-         eosio::abi_def abidef = eosio::from_json<eosio::abi_def>(token_stream);
-         eosio::convert(abidef, abi);
+         std::string abi_string = boost::beast::buffers_to_string(abi_buffer.data());
+         //remove all tables since their names are invalid; tables not needed for this test
+         std::regex scrub_all_tables(R"(\{ "name": "[^"]+", "type": "[^"]+", "key_names": \[[^\]]*\] \},?)");
+         abi_string = std::regex_replace(abi_string, scrub_all_tables, "");
+
+         abi = eosio::chain::abi_serializer(fc::json::from_string(abi_string).as<eosio::chain::abi_def>(), null_yield_function);
+         //state history may have 'bytes' larger than MAX_SIZE_OF_BYTE_ARRAYS, so divert 'bytes' to an impl that does not have that check
+         abi.add_specialized_unpack_pack("bytes", std::make_pair<eosio::chain::abi_serializer::unpack_function, eosio::chain::abi_serializer::pack_function>(
+            [](fc::datastream<const char*>& stream, bool is_array, bool is_optional, const eosio::chain::abi_serializer::yield_function_t& yield) {
+               FC_ASSERT(!is_array, "sorry, this kludge doesn't support arrays");
```

Comment:
`bytes` is a `vector<char>`, so shouldn't this assert say "bytes can't be an array" or something like that?

Reply:
yeah  I guess I can change it to `doesn't support bytes[]` to be more specific

Reply:
Oh is `bytes[]` a supported type? I have used this streamer with production ship servers as a way of testing. If that is something the ship data can contain then it would be nice to support it.

Reply:
Yes `bytes[]` is supported in ABIs

But you'll never encounter it in the response here since ship_streamer only parses (and then dumps out) the get_block_result i.e.
https://github.com/AntelopeIO/spring/blob/808f061b89a080fbf60cc0636ca0b38b134e523a/libraries/state_history/abi.cpp#L81-L92
(so `bytes?` did need support). ship_streamer doesn't do any deeper inspection of the block, traces, etc

Reply:
Nice. Thanks.